src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -77,8 +94,10 @@ public void StartFileChangeListeningAsync()
                 return newCookie;
             }, LazyThreadSafetyMode.ExecutionAndPublication);
 
-            // file change service is free-threaded. start running it in background right away
-            Task.Run(() => _fileChangeCookie.Value, CancellationToken.None);
+            lock (s_lastBackgroundTaskGate)
+            {
+                s_lastBackgroundTask = s_lastBackgroundTask.ContinueWith(_ => _fileChangeCookie.Value, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
```

Comment:
@CyrusNajmabadi Looking at the one trace @Pilchie provided with this fix, of the 3.57 seconds being spent in FileChangeTracker, 3.44 was being spent in the service itself, so our overhead is quite minor. And the overhead is definitely smaller than the other bits of silliness we've identified. I suspect we're better off junking this than trying to get fancy.